src/org/usfirst/frc/team2706/robot/commands/StraightDriveWithTime.java

```diff
@@ -0,0 +1,71 @@
+package org.usfirst.frc.team2706.robot.commands;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.usfirst.frc.team2706.robot.Robot;
+import edu.wpi.first.wpilibj.command.Command;
+
+/**
+ * Have the robot drive certain amount of time
+ */
+public class StraightDriveWithTime extends Command {
+    
+	private final double speed;
+	
+	private final long time;
+	
+	/**
+	 * Drive at a specific speed for a certain amount of time
+	 * 
+	 * @param speed Speed in range [-1,1]
+	 * @param time Time in milliseconds to drive
+	 */
+    public StraightDriveWithTime(double speed, long time) {
+        requires(Robot.driveTrain);
+
+        this.speed = speed;
+        
+        this.time = time;
+    }
+
+    // Called just before this Command runs the first time
+    protected void initialize() {
+    	StraightDriveWithTime ref = this;
```

Comment:
Write a new wrapper class for the TimerTask and pass "this":

class CommandTimerTask extends TimerTask {
    Command command;
    public CommandTimerTask(Command command) {
        super();
        this.command = command;
    }
    public void run() {
        command.cancel();
    }
}

// calling the above
CommandTimerTask interrupt = new CommandTimerTask(this);
new Timer().schedule(interrupt, time);